### PR TITLE
Simplify update handling as per consistent filtering proposal

### PR DESCRIPTION
**Pull request by RReverser**
## Description of Changes

As a result of the consistent filtering rules proposal (clockworklabs/SpacetimeDB#1256), all equatable types are now natively equatable in C# as well. 

This allows us to compare objects directly, without holding and comparing AlgebraicValue in the entries map as well, which has a bit of a domino effect and allows to optimise, simplify or even remove some parts of the SDK.

## API

 - [x] This is an API breaking change to the SDK

New filtering rules limit types on which filtering can be done, as well as change the return type of `FilterBy` functions to always be iterable.

## Requires SpacetimeDB PRs

https://github.com/clockworklabs/SpacetimeDB/pull/1277

### CODE REVIEW

File: examples/quickstart/client/module_bindings/Message.cs [hunk 1]
```diff
@@ -43,9 +43,7 @@ public static SpacetimeDB.SATS.AlgebraicType GetAlgebraicType()
 
 		public static explicit operator Message(SpacetimeDB.SATS.AlgebraicValue value)
 		{
-			if (value == null) {
-				return null;
-			}
```
Comment (jdetter) [inline examples/quickstart/client/module_bindings/Message.cs:48]: I know its somewhat controvertial but our style guide typically requires the braces here

**Review by jdetter — APPROVED** [review]
Small nit but overall this is a good refactor. I have not tested it myself but will be doing an overall test of the C#/Unity SDK later this week so this should get tested as part of that.

File: examples/quickstart/client/module_bindings/Message.cs (hunk 1 above)
Comment (RReverser) [reply examples/quickstart/client/module_bindings/Message.cs:48]: I think it got changed in one of the previous PRs, it's just that module bindings weren't regenerated. If we want to revert, probably worth doing outside of this PR.

Comment (RReverser) [reply examples/quickstart/client/module_bindings/Message.cs:48]: Ah, I'm actually going to remove that function altogether in the next PR, so hopefully that solves it too.

**Pull request merged**